src/mono/mono/component/debugger-agent.c

```diff
@@ -434,6 +434,17 @@ static gboolean buffer_replies;
 	tls = &debugger_wasm_thread;
 #endif
 
+#define GET_EXTRA_SPACE_FOR_REF_FIELDS(klass) \
+	extra_space_size = 0; \
+	extra_space = NULL; \
+	if (m_class_is_valuetype (klass) && (m_type_is_byref (m_class_get_byval_arg (klass)) || m_class_is_byreflike (klass))) { \
```

Comment:
`m_class_is_valuetype (klass) && (m_type_is_byref (m_class_get_byval_arg (klass))` can't be true in mono right now.  the byval_arg of every MonoClass is not a ref type.

(In my `ref ref C` prototype from hack week a couple years ago, we added a MonoClass for byref types, so this might be ok for future-proofing, but right now it won't do anything).

Probably you need a MonoType here